chsarp/DefaultLambdaParameters.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.Extensions.Logging;
+
+namespace CSharp;
+
+public class DefaultLambdaParameters(ILogger<CollectionExpressions> _logger)
+{
+    public void DefaultParameters()
+    {
+        var incrementValue = (int value = 2) =>
+        {
+            _logger.LogInformation($"{(value == 2 ? "Called with default value" : "Called with given value")}: {value}");
+
+            return value + 1;
+        };
+
+        incrementValue();
+        incrementValue(5);
+
+        var action = incrementValue;
+        action();
```

Comment:
I think it's unnecessary since you're already calling twice above.

Reply:
i agree. if it showcases something else, you need to be clear about it.

Reply:
This is to showcase the new behaviour which previously gave error due to a mismatch in the number of expected arguments. https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-12.0/lambda-method-group-defaults#current-behavior

Reply:
nice :ok_hand: 

then don't imply it, be explicit. you can rename `action` to something else to be explicit.